crates/op-rbuilder/src/payload_builder_vanilla.rs

```diff
@@ -1104,6 +1134,30 @@ where
                     PayloadBuilderError::other(OpPayloadBuilderError::TransactionEcRecoverFailed)
                 })?;
 
+            // op-supervisor validation
+            if let Some(access_list) = sequencer_tx.access_list() {
+                match self.validate_supervisor_messages(access_list)? {
+                    Ok(()) => (),
+                    Err(err) => match err {
+                        InteropTxValidatorError::SupervisorServerError(err) => {
+                            warn!(target: "payload_builder", %err, ?sequencer_tx, "Supervisor error, skipping.");
+                            self.metrics.inc_num_cross_chain_tx_server_error();
+                            continue;
+                        }
+                        InteropTxValidatorError::ValidationTimeout(_) => {
+                            trace!(target: "payload_builder", %err, ?sequencer_tx, "Executing message validation timed out, skipping.");
+                            self.metrics.inc_num_cross_chain_tx_timeout();
+                            continue;
+                        }
+                        err => {
+                            trace!(target: "payload_builder", %err, ?sequencer_tx, "Executing message rejected.");
+                            self.metrics.inc_num_cross_chain_tx_fail();
+                            continue;
+                        }
+                    },
+                }
```

Comment:
there is also the client error which you probably want to match on specifically, because depending on what it is, you may want to try and reconnect with the supervisor